Tests/NIOCoreTests/ByteBufferQUICBinaryEncodingStrategyTests.swift

```diff
@@ -0,0 +1,136 @@
+    // MARK: - writeEncodedInteger tests
+
+    func testWriteOneByteQUICVariableLengthInteger() {
+        // One byte, ie less than 63, just write out as-is
+        for number in 0..<63 {
+            var buffer = ByteBuffer()
+            let strategy = ByteBuffer.QUICBinaryEncodingStrategy.quic
+            let bytesWritten = strategy.writeInteger(number, to: &buffer)
+            XCTAssertEqual(bytesWritten, 1)
+            // The number is written exactly as is
+            XCTAssertEqual(buffer.readInteger(as: UInt8.self), UInt8(number))
+            XCTAssertEqual(buffer.readableBytes, 0)
+        }
+    }
+
+    func testWriteBigUInt8() {
+        // This test case specifically tests the scenario where 2 bytes are needed, but the number being written is UInt8.
+        // A naive implementation of the quic variable length integer encoder might check whether the number is in
+        // the range of 64..<16383, to determine that it should be written with 2 bytes.
+        // However, constructing such a range on a UInt8 would actually construct 64..<0, because 16383 can't be represented as UInt8.
+        // So this test makes sure we didn't make that mistake
+        let number: UInt8 = .max
+        var buffer = ByteBuffer()
+        let strategy = ByteBuffer.QUICBinaryEncodingStrategy.quic
+        let bytesWritten = strategy.writeInteger(number, to: &buffer)
+        XCTAssertEqual(bytesWritten, 2)
+        XCTAssertEqual(buffer.readInteger(as: UInt16.self), 0b01000000_11111111)
+        XCTAssertEqual(buffer.readableBytes, 0)
+    }
+
+    func testWriteTwoByteQUICVariableLengthInteger() {
+        var buffer = ByteBuffer()
+        let strategy = ByteBuffer.QUICBinaryEncodingStrategy.quic
+        let bytesWritten = strategy.writeInteger(0b00111011_10111101, to: &buffer)
+        XCTAssertEqual(bytesWritten, 2)
+        // We need to mask the first 2 bits with 01 to indicate this is a 2 byte integer
+        // Final result 0b01111011_10111101
+        XCTAssertEqual(buffer.readInteger(as: UInt16.self), 0b01111011_10111101)
+        XCTAssertEqual(buffer.readableBytes, 0)
+    }
+
+    func testWriteFourByteQUICVariableLengthInteger() {
+        var buffer = ByteBuffer()
+        let strategy = ByteBuffer.QUICBinaryEncodingStrategy.quic
+        let bytesWritten = strategy.writeInteger(0b00011101_01111111_00111110_01111101, to: &buffer)
+        XCTAssertEqual(bytesWritten, 4)
+        // 2 bit mask is 10 for 4 bytes so this becomes 0b10011101_01111111_00111110_01111101
+        XCTAssertEqual(buffer.readInteger(as: UInt32.self), 0b10011101_01111111_00111110_01111101)
+        XCTAssertEqual(buffer.readableBytes, 0)
+    }
+
+    func testWriteEightByteQUICVariableLengthInteger() {
+        var buffer = ByteBuffer()
+        let strategy = ByteBuffer.QUICBinaryEncodingStrategy.quic
+        let bytesWritten = strategy.writeInteger(
+            0b00000010_00011001_01111100_01011110_11111111_00010100_11101000_10001100,
```

Comment:
@hamzahrmalik, this line and a couple more writes below [broke compilation of these tests for 32-bit platforms like Android armv7 on my Android CI](https://github.com/finagolfin/swift-android-sdk/actions/runs/11159109574/job/31016848537#step:15:6552).

Reply:
@finagolfin @Lukasa @FranzBusch can we integrate the Android & 32 bit CIs into NIO's own github actions?

Reply:
Franz and I discussed getting Android tested on this CI, I'm aiming to have something for that in the coming months. However, that just builds the source for armv7 but wouldn't run the tests.

If you have a 32-bit CI like watchOS or embedded that you could add quickly, that may be a better first step to get 32-bit platforms tested.

Reply:
I've raised a PR to fix this: #2904